src/ImageProcessor.Web/Extensions/StringExtensions.cs

```diff
@@ -138,6 +138,17 @@ public static bool IsValidVirtualPathName(this string expression)
 
             return Uri.TryCreate(expression, UriKind.Relative, out uri) && uri.IsWellFormedOriginalString();
         }
+        /// <summary>
+        /// As url segments cannot contain : it will be replaced with _
+        /// However, we need the correct url for requesting local host on port 10000 where 
+        /// Development Storage Emulator is listening
+        /// </summary>
+        /// <param name="url">The <see cref="T:System.String">String</see> instance that this method extends.</param>
+        /// <returns>returns corrected url with host:port</returns>
+        public static string FixDevelopmentStorageEmulatorUrl(this string url)
```

Comment:
I'm not completely happy adding something like this. By all means we should be able to use a url with a port but that should be something that is url encoded by the dev and then decoded in the HttpModule for IImageService implementations to use.  

I've also used this url/port combination before for hosting projects I have in development in IIS so I don't want potential conflict.